lib/internal/modules/esm/translators.js

```diff
@@ -33,6 +35,19 @@ const debug = debuglog('esm');
 const translators = new SafeMap();
 exports.translators = translators;
 
+const DATA_URL_PATTERN = /^[^/]+\/[^,;]+(;base64)?,([\s\S]*)$/;
+function getSource(url) {
+  const parsed = new URL(url);
+  if (parsed.protocol === 'file:') {
+    return readFileAsync(parsed);
+  } else if (parsed.protocol === 'data:') {
+    const [ , base64, body ] = DATA_URL_PATTERN.exec(parsed.pathname);
```

Comment:
Should we restrict the size? Looks like modern browsers don't (in any meaningful way).

Reply:
Is there a reason to?

Reply:
@jkrems since we aren't parsing and just reading until the end for characters I doubt a catastrophic backtracking DOS could occur.